diff -ur Ice-3.5.1-orig/cpp/Makefile Ice-3.5.1/cpp/Makefile
--- Ice-3.5.1-orig/cpp/Makefile	2013-10-07 14:19:37.712388988 +0100
+++ Ice-3.5.1/cpp/Makefile	2013-10-07 14:20:08.704323464 +0100
@@ -11,11 +11,7 @@
 
 include $(top_srcdir)/config/Make.rules
 
-SUBDIRS		= config src include test
-
-ifeq ($(shell uname | grep MINGW),)
-SUBDIRS		:= $(SUBDIRS) demo
-endif
+SUBDIRS		= config src include
 
 INSTALL_SUBDIRS	= $(install_bindir) $(install_libdir) $(install_includedir) \
 	$(install_configdir) $(install_mandir)
diff -ur Ice-3.5.1-orig/cs/Makefile Ice-3.5.1/cs/Makefile
--- Ice-3.5.1-orig/cs/Makefile	2013-10-07 14:19:37.756388895 +0100
+++ Ice-3.5.1/cs/Makefile	2013-10-07 14:20:23.200292801 +0100
@@ -11,7 +11,7 @@
 
 include $(top_srcdir)/config/Make.rules.cs
 
-SUBDIRS		= src test demo
+SUBDIRS		= src
 
 INSTALL_SUBDIRS	= $(install_bindir) $(install_assembliesdir) $(install_mandir)
 
diff -ur Ice-3.5.1-orig/java/Makefile Ice-3.5.1/java/Makefile
--- Ice-3.5.1-orig/java/Makefile	2013-10-07 14:19:37.619389185 +0100
+++ Ice-3.5.1/java/Makefile	2013-10-07 14:20:47.896240537 +0100
@@ -24,7 +24,7 @@
 endif
 
 all:
-	ant $(ANT_FLAGS)
+	ant $(ANT_FLAGS) dist-jar
 
 clean:
 	ant $(ANT_FLAGS) clean
