--- ice-3.8.1/config/Make.rules.Linux~	2026-04-20 18:57:56.000000000 +0200
+++ ice-3.8.1/config/Make.rules.Linux	2026-04-20 19:01:20.096672176 +0200
@@ -35,14 +35,14 @@
 # Targeting x86-64-v2, compatible with most x86-64 processors. This matches the defaults in RHEL 9 x86-64 builds.
 # https://developers.redhat.com/blog/2021/01/05/building-red-hat-enterprise-linux-9-for-the-x86-64-v2-microarchitecture-level
 
-x64_cppflags            := $(call opt-cppflags,x86_64) -m64 -march=x86-64-v2
-x64_ldflags             := $(call opt-ldflags,x86_64) -m64
+x64_cppflags            := $(call opt-cppflags,x86_64)
+x64_ldflags             := $(call opt-ldflags,x86_64)
 x64_targetdir           = $(if $(filter %/build,$5),/x64,$(if $(filter-out $($1_target),program),64))
 x64_installdir          = $(if $(and $(filter-out $($1_target),program),$(if $5,$(filter-out %64,$5),true)),64)
 x64_cmakedir            = /cmake
 
-x86_cppflags            := $(call opt-cppflags,i386) -m32 -march=x86-64-v2
-x86_ldflags             := $(call opt-ldflags,i386) -m32
+x86_cppflags            := $(call opt-cppflags,i386)
+x86_ldflags             := $(call opt-ldflags,i386)
 x86_targetdir           = $(if $(filter %/build,$5),/x86)
 x86_targetname          = $(if $(is-bin-program),32)
 x86_cmakedir            = /cmake
