--- ice-3.6.3/cpp/config/Make.rules.Linux.orig	2016-10-05 16:59:08.000000000 +0200
+++ ice-3.6.3/cpp/config/Make.rules.Linux	2017-06-28 16:32:48.328625292 +0200
@@ -79,18 +79,6 @@
         CXXARCHFLAGS	+= -mtune=v8 -pipe -Wno-deprecated -DICE_USE_MUTEX_SHARED
     endif
 
-    #
-    # Only use multilib -m32/-m64 options for amd64 arch, x32 arch can also run
-    # in amd64 macines but doesn't use multilib, in this case MACHINE_TYPE will
-    # be x86_64 and HOST_ARCH x32.
-    #
-    ifeq ($(HOST_ARCH),amd64)
-        ifeq ($(LP64),yes)
-            CXXARCHFLAGS	+= -m64
-        else
-            CXXARCHFLAGS	+= -m32
-        endif
-    endif
 
     CXXFLAGS		+= $(CXXARCHFLAGS) -fvisibility=hidden -Wall -Werror -pthread
 
--- ice-3.6.3/config/Make.common.rules.orig	2016-10-05 16:59:08.000000000 +0200
+++ ice-3.6.3/config/Make.common.rules	2017-06-28 16:40:31.135286671 +0200
@@ -102,24 +102,8 @@
           LP64      	= yes
       endif
 
-      #
-      # x86 binaries build on x86_64 machine use 32 binsuffix
-      #
-      ifneq ($(STATICLIBS),yes)
-          ifneq ($(LP64),yes)
-              binsuffix	= 32
-          endif
-      endif
    endif
 
-   #
-   # x86 builds uses 32 binsuffix
-   #
-   ifneq ($(STATICLIBS),yes)
-	ifeq ($(findstring $(MACHINE_TYPE),i386 i486 i586 i686),$(MACHINE_TYPE))
-	    binsuffix     = 32
-	endif
-    endif
 endif
 
 ifeq ($(CPP11),yes)
