--- ice-3.8.1/config/Make.rules.Linux~	2026-03-02 16:35:34.000000000 +0100
+++ ice-3.8.1/config/Make.rules.Linux	2026-04-20 16:21:10.766669622 +0200
@@ -140,7 +140,7 @@
 # Link the Slice compilers without any shared libraries.
 # On Amazon Linux 2023, disable static linking since libmcpp-devel provides only the shared library.
 ifeq ($(filter amzn,$(linux_id)),)
-fully_static_ldflags = -static
+#fully_static_ldflags = -static
 endif
 
 cppflags        = -fvisibility=hidden -Wall -Wextra -Wredundant-decls -Wshadow -Wdeprecated \
